### ISSUE TITLE
How to add sample tilt in phaser?

### ISSUE THREAD

**Issue by xielin1984** [hexane360/phaser#47]
Hi @hexane360 ,
    Thanks for this great software. I have run into problems in adding sample tilt in the reconstruction because I cannot find any example or tutorial. I tried to add "tilt_hook" in json file, but I got "WARNING:phaser.engines.common.output:Tilt map (`state.tilt`) is missing, skipping `plot_tilt`". Any help is appreciated. Thanks

Best wishes,

**Comment by hexane360** [comment]
To initialize a uniform tilt map, you can use this:
```yaml
init:
  tilt:
    type: global
    tilt: [0.0, 0.0]  # [y, x] in mrad (+ is down/right)
```

Then, in your reconstruction solvers you can add a solver for the `tilt` variable, and a tilt regularization:
```yaml
engines:
  - type: 'gradient'
    ...
    solvers:
      ...
      tilt:
        type: sgd
        learning_rate: 50.0  # will have to tune this to give reasonable update magnitudes
        momentum: 0.5
        nesterov: True

      iter_constraints:
        ...
        - type: tilt_gaussian
          sigma: 1.5   # angstrom
          weight: 0.1 
```
The `tilt_gaussian` regularization blurs together tilts from nearby probe positions, to prevent atomic structure from ending up in the reconstructed tilts.

Also, I'd recommend using the `develop` branch for now, it's more up to date. I'm currently working on pushing a new stable release.

Let's keep this issue open until we include a better example file for tilt correction.

**Comment by xielin1984** [comment]
Hi @hexane360 ,
   Great!  Looking forward to a new stable release and the example. For my question, I am using the LSQML algorithm and comparing to fold_slice right now. Is there a way to update tilt as well?

Best regards,

**Comment by hexane360** [comment]
Currently the sample tilt update is not supported for the conventional solvers (ePIE and LSQML). However, existing tilts from other engines are respected. You may be able to use a LSQML engine followed by a gradient descent engine that mainly updates tilts, or LSQML -> gradient -> LSQML.

**Comment by xielin1984** [comment]
Hi @hexane360 ,
   OK. I am now testing the hyperparameters systematically. One more question, can I get a print of tilt on screen after the reconstruction?

Best regards

**Comment by hexane360** [comment]
You can add 'tilt' to the list in `save_options.images` to save a tilt map. Also, `plot_ext` and `plot_dpi` in `save_options` can be used to control the plot output
